services/traction/api/tasks/tasks.py

```diff
@@ -207,6 +216,27 @@ async def _perform_task(self, tenant: Tenant, payload: dict):
             self.logger.error()
         self.logger.info("< _perform_task()")
 
+    async def _handle_perform_task_error(
+        self, tenant: Tenant, payload: dict, exc: Exception
+    ):
+        self.logger.info("> _handle_perform_task_error()")
+        schema_template_id = payload["schema_template_id"]
+        values = {"status": "Error", "error_status_detail": str(exc)}
+        q = (
+            update(SchemaTemplate)
+            .where(SchemaTemplate.schema_template_id == schema_template_id)
+            .values(values)
+        )
+        async with async_session() as db:
+            try:
+                await db.execute(q)
+            except DBAPIError:
+                await db.rollback()
+                self.logger.error(exc_info=1)
+            else:
+                await db.commit()
```

Comment:
this chunk of code is gonna show up alot.... can this be a class method on the db class? just accepts a dict of {column:new_value}

we aren't really using the ORM at all, but this is a blind update, so i see the benefit.

Reply:
It would also be great to see this generalized and centralised to the `StatefulModel` class... could use some reflection to find the pk column and use that id to update the row. I've written some reflection code I can point you to.

Reply:
could do either way.

Reply:
set a utility function in base model, its an encapsulated read/update transaction/session.